### PR TITLE
gensis-config: patching default `RuntimeGenesisConfig` fixed

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR  changes the behavior of  `json_patch::merge` function which no longer removes any keys from the base JSON object.

fixes: #6306

### CODE REVIEW

**Comment by bkchr** [comment]
Do we really need these two functions? IMO `null` should not be handled differently than any other kind of value. If someone gives such a json, just accept it. 

TLDR: I would just fix the `merge` function and not introduce a new one.

**Comment by bkchr** [comment]
> Idea of removing `null` comes from this document https://www.rfc-editor.org/rfc/rfc7386

Yeah I saw this. 



> I just don't want to break downstream which potentially can use `json_patch::merge` for actually removing fields. I think proposed approach is "safest" from this point of view..

I mean already break downstream by removing the function export from `sc_chain_spec` :see_no_evil: I would still vote for simplicity. Question is how many people downstream are using the field removal. We would also need to change the chain-spec-generator to not remove the fields, which is also a behavior change in the program. Generally, I think that merging the null values into the final json makes more sense any way.

File: prdoc/pr_6382.prdoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    This PR fixes issue reported in #6306.
+    It changes the behavior of `sc_chain_spec::json_patch::merge` function which no longer removes any keys from the base JSON object.
+
+crates:
+- name: staging-chain-spec-builder
+  bump: major
+- name: sc-chain-spec
+  bump: major
+- name: frame-benchmarking-cli
+  bump: major
```
Comment (michalkucharczyk) [inline prdoc/pr_6382.prdoc:14]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Review by dmitry-markin — APPROVED** [review]

File: substrate/bin/utils/chain-spec-builder/src/lib.rs [hunk 2]
```diff
@@ -261,19 +261,23 @@ impl ChainSpecBuilder {
...
+				// just remove it:
+				org_chain_spec_json.as_object_mut().map(|map| {
 					map.get_mut("genesis").map(|genesis| {
-						genesis.as_object_mut().map(|genesis_map| {
-							genesis_map
-								.insert("runtimeGenesis".to_string(), serde_json::Value::Null);
-						});
+						genesis
+							.as_object_mut()
+							.map(|genesis_map| genesis_map.remove("runtimeGenesis"));
```
Comment (iulianbarbu) [inline substrate/bin/utils/chain-spec-builder/src/lib.rs:277]: nit:  Would something like this work? arguably looks cleaner

```rust
org_chain_spec_json
   .get_mut("genesis")
   .and_then(|inner| inner.as_object_mut())
   .map(|inner| inner.remove("runtimeGenesis"));
```

**Review by iulianbarbu — APPROVED** [review]

File: substrate/bin/utils/chain-spec-builder/src/lib.rs (hunk 2 above)
Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs]: c95cfa1

Comment (michalkucharczyk) [reply substrate/bin/utils/chain-spec-builder/src/lib.rs]: yeah, even better: 62a780f

**Review by lexnv — APPROVED** [review]

**Pull request merged**